### PR TITLE
Add CURL handler last error logging

### PR DESCRIPTION
**Pull request by serge-webspark**
Hi there! I improved the CURL request logging by adding the last error code and message. It would be useful for such cases as certificate expiration

```
Error number: 56, Error message: OpenSSL SSL_read: error:14094415:SSL routines:ssl3_read_bytes:sslv3 alert certificate expired, errno 0.
```

Thank you in advance!

### CODE REVIEW

File: ApnsPHP/Push.php [hunk 1]
```diff
@@ -638,6 +638,12 @@ private function httpSend(Message $message, &$reply): bool
             CURLOPT_POSTFIELDS => $message->getPayload()
             ]) && ($reply = curl_exec($this->hSocket)) !== false)
         ) {
+            $errorCode = curl_errno($this->hSocket);
+
+            if ($errorCode > 0) {
+                $reply = sprintf('Response: %s; Error code: %d; Error message: %s.', $reply, $errorCode, curl_error($this->hSocket));
```
Comment (SMillerDev) [inline ApnsPHP/Push.php:644]: Shouldn't this use the logger? I don't see how `$reply` can be used any further.

Comment (serge-webspark) [reply ApnsPHP/Push.php:644]: We pass the `$reply` parameter by reference (&) and then use it in error log 
https://github.com/move-backend/ApnsPHP/blob/master/ApnsPHP/Push.php#L577

Comment (SMillerDev) [reply ApnsPHP/Push.php:644]: Ah you're right. `$reply` seems to have a very specific meaning though, and I don't think we should use it for general logging. Let's just use the logger here.

Comment (serge-webspark) [reply ApnsPHP/Push.php]: We just pass it to `statusMessage` log field but okay replaced the reply usage with the logger call.

File: ApnsPHP/Push.php [hunk 2]
```diff
@@ -638,6 +638,12 @@ private function httpSend(Message $message, &$reply): bool
             CURLOPT_POSTFIELDS => $message->getPayload()
             ]) && ($reply = curl_exec($this->hSocket)) !== false)
         ) {
+            $errorCode = curl_errno($this->hSocket);
+
+            if ($errorCode > 0) {
+                $this->logger->error(sprintf('HTTP request error code: %d; message: %s.', $errorCode, curl_error($this->hSocket)));
```
Comment (SMillerDev) [inline ApnsPHP/Push.php:644]: Let's use the logger context here:
```suggestion
                $this->logger->error('HTTP request error code: {code}; message: {message}.', ['code' => $errorCode, 'message' => curl_error($this->hSocket)]);
```

Comment (serge-webspark) [reply ApnsPHP/Push.php]: Added context

**Comment by serge-webspark** [comment]
Hi! Any plans to merge the changes?

**Comment by serge-webspark** [comment]
Just noticed it, resolved the issue.

**Review by SMillerDev — APPROVED** [review]

**Comment by pprkut** [comment]
@serge-webspark Thank you! :slightly_smiling_face:

**Pull request closed without merging**